tests/testthat/test-pkg-dplyr.R

```diff
@@ -225,3 +225,129 @@ test_that("relocate() for substrait_compiler reorders columns", {
     substrait_select(compiler, col2, col1)
   )
 })
+
+test_that("all types of dplyr mutating joins can be created from substrait_comiler", {
+  skip_if_not_installed("dplyr")
+
+  df_left <- data.frame(number = 1:26, letter = letters, stringsAsFactors = FALSE)
+  df_right <- data.frame(number = 1:26, LETTER = LETTERS, stringsAsFactors = FALSE)
+  left <- join_dummy_compiler(df_left)
+
+  # Semi join currently implemented as inner join + emit because support for that
+  # appears to be better.
+  joined_semi <- dplyr::semi_join(left, df_right)
+  expect_s3_class(joined_semi, "SubstraitCompiler")
+  expect_identical(
+    joined_semi$rel$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_INNER)
+  )
+  expect_identical(
+    joined_semi$schema$names,
+    names(df_left)
+  )
+
+  joined_anti <- dplyr::anti_join(left, df_right)
+  expect_s3_class(joined_anti, "SubstraitCompiler")
+  expect_identical(
+    joined_anti$rel$project$input$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_ANTI)
+  )
+  expect_identical(
+    joined_anti$schema$names,
+    names(df_left)
+  )
+})
+
+test_that("all types of dplyr mutating joins can be created from substrait_comiler", {
+  skip_if_not_installed("dplyr")
+
+  df_left <- data.frame(number = 1:26, letter = letters, stringsAsFactors = FALSE)
+  df_right <- data.frame(number = 1:26, LETTER = LETTERS, stringsAsFactors = FALSE)
+  left <- join_dummy_compiler(df_left)
+
+  joined_left <- dplyr::left_join(left, df_right)
+  expect_s3_class(joined_left, "SubstraitCompiler")
+  expect_identical(
+    joined_left$rel$project$input$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_LEFT)
+  )
+
+  joined_right <- dplyr::right_join(left, df_right)
+  expect_s3_class(joined_left, "SubstraitCompiler")
+  expect_identical(
+    joined_right$rel$project$input$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_RIGHT)
+  )
+
+  joined_inner <- dplyr::inner_join(left, df_right)
+  expect_s3_class(joined_left, "SubstraitCompiler")
+  expect_identical(
+    joined_inner$rel$project$input$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_INNER)
+  )
+
+  joined_full <- dplyr::full_join(left, df_right)
+  expect_s3_class(joined_left, "SubstraitCompiler")
+  expect_identical(
+    joined_full$rel$project$input$join$type,
+    as.integer(substrait$JoinRel$JoinType$JOIN_TYPE_OUTER)
+  )
+})
+
+test_that("dplyr mutating joins for substrait_compiler support by, keep, and suffix", {
```

Comment:
How about we group all of the tests for the mutating joins together, i.e. grouped by function instead of parameter set?